### PR TITLE
DOC - add ``Whatsnew`` section

### LINKED ISSUE #71: DOC add Whatsnew.rst

### PR DESCRIPTION
**Pull request by PABannier**
Since `skglm` is growing bigger, we need to keep track of contributions and API changes in the library. 
This PR introduces a `Whatsnew` section with a `changes` folder in the `doc`.
I got inspired by [MNE-Python](https://github.com/mne-tools/mne-python) template. WDYT?

Closes #71

### CODE REVIEW

**Review by Badr-MOUFAD — COMMENTED** [review]
Thanks @PABannier for this PR.
I have one concern. These changes were (presumably) written manually. I wonder how to keep the whats_new synchronized with the current changes in the package. For instance here, I see two missing major changes:
- Group solver
- API changes in solvers (function --> class)

**Comment by mathurinm** [comment]
Good point. In sklearn and benchopt, we do it manually which makes me think they're no easy solution

Here it was costly to do because we had to track back all changes. Now we only need to require it for every merged PR, it'll be easier.

**Comment by PABannier** [comment]
Yup agreed. I was off for a month, so I probably forgot a few additions.
In the future, for every major PR, we should require to document changes in the changelog file. That's how it's done in MNE-Python.

File: doc/sphinxext/gh_substitutions.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+"""Provide a convenient way to link to GitHub issues and pull requests.
```
Comment (Badr-MOUFAD) [inline doc/sphinxext/gh_substitutions.py:1]: Nice reflex @mathurinm to group the sphinx py modules in a separate folder.
We should consider doing the same for ``github_link.py`` (in another PR)

**Review by Badr-MOUFAD — COMMENTED** [review]
Just a small point to discuss before merging

File: doc/changes/0.1.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
+.. _changes_0_1:
+
+Version 0.1 (2022-04-25)
+------------------------
+
+- Initial release (initial commit: 2022-04-20)
+
+- Accelerated coordinate descent solver with a working set strategy
```
[the remark this answers is not in the document]
Comment (PABannier) [reply doc/changes/0.1.rst:8]: Agreed

[the remark this answers is not in the document]
Comment (mathurinm) [reply doc/changes/0.1.rst:8]: agreed too !

**Review by Badr-MOUFAD — APPROVED** [review]
LGTM!

File: doc/changes/0.2.rst [hunk 3]
```diff
@@ -0,0 +1,18 @@
+.. _changes_0_2:
+
+Version 0.2 (in progress)
+------------------------
+
+- Accelerated block coordinate descent solver with working sets for problems with group penalties (:gh:`29`, :gh:`28`, :gh:`26`)
+
+- Proximal Newton solver with working sets for problems with non-quadratic datafits (:gh:`51`)
```
Comment (mathurinm) [inline doc/changes/0.2.rst:8]: can you put the name of the class here (as for datafits?). It guides the user to the correct class

File: doc/changes/0.2.rst [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+- ``Huber`` datafit (:gh:`14`)
+
+- SCAD and Block SCAD penalties supported (:gh:`12`)
```
Comment (mathurinm) [inline doc/changes/0.2.rst:12]: class names here too

File: doc/changes/0.2.rst [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+- Fitting intercept for ``Quadratic``, ``Logistic``, ``Huber`` datafits (:gh:`55`)
+
+- Gram coordinate descent solver (:gh:`59`)
```
Comment (mathurinm) [inline doc/changes/0.2.rst:16]: class name here too

File: doc/changes/0.2.rst (hunk 3 above)
Comment (mathurinm) [reply doc/changes/0.2.rst:8]: even better if you manage to link to the doc of the corresponding class...

**Review by mathurinm — CHANGES_REQUESTED** [review]

File: doc/changes/0.2.rst (hunk 3 above)
Comment (mathurinm) [reply doc/changes/0.2.rst]: we should :)

**Comment by mathurinm** [comment]
I changed the way links are formatted; Scad, BlockSCAD and GramCD don't appear because they are missing from the doc.

@Badr-MOUFAD could you add them to `api.rst`, with all the other solvers?

**Comment by mathurinm** [comment]
Links for scad and blockscad are not working (+ I get warnings when running `make html-noplot`)

When this is fixed, go for merge!

**Comment by Badr-MOUFAD** [comment]
They don't appear as they are not included in the API ...

**Comment by mathurinm** [comment]
so, they should be included in the API. it can be done in this PR

**Comment by mathurinm** [comment]
thank you all

**Pull request merged**